src/sentry/llm/providers/base.py

```diff
@@ -0,0 +1,45 @@
+from sentry.llm.exceptions import InvalidModelError, InvalidProviderError
+from sentry.llm.types import ProviderConfig, UseCaseConfig
+from sentry.utils.services import Service
+
+
+class LlmModelBase(Service):
+    def __init__(self, provider_config: ProviderConfig) -> None:
+        self.provider_config = provider_config
+
+    def complete_prompt(
+        self,
+        *,
+        usecase_config: UseCaseConfig,
+        prompt: str,
+        message: str,
+        temperature: float,
+        max_output_tokens: int,
+    ) -> str | None:
+        self.validate_model(usecase_config["options"]["model"])
+
+        return self._complete_prompt(
+            usecase_config=usecase_config,
+            prompt=prompt,
+            message=message,
+            temperature=temperature,
+            max_output_tokens=max_output_tokens,
+        )
+
+    def _complete_prompt(
+        self,
+        *,
+        usecase_config: UseCaseConfig,
+        prompt: str,
+        message: str,
```

Comment:
I would make it clearer that the prompt is the system message as it holds a special role in most LLM models and is treated differently. Also, it's optional and not required, only the initial user message is.